### PR TITLE
Add new rustdoc book chapter to describe in-doc settings

### LINKED ISSUE #55165: add a section to the Rustdoc Book about the in-doc settings
Rustdoc has accumulated a handful of settings that users can toggle as they read docs, but we don't document them anywhere. It would be ideal if we had some documentation about these options in the Rustdoc Book that described their functionality better than the short descriptions shown on the settings page itself.

image

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #55165.

I continue going through old rustdoc issues. This one made a lot of sense so decided to add the missing chapter.

r? @notriddle

### CODE REVIEW

**Review by notriddle — CHANGES_REQUESTED** [review]

File: src/doc/rustdoc/src/in-doc-settings.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Rustdoc in-doc settings
+
+Rustdoc provides settings alongside the theme picker to personalize your
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/doc/rustdoc/src/in-doc-settings.md:3]: Agreed.

File: src/doc/rustdoc/src/SUMMARY.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 - [What is rustdoc?](what-is-rustdoc.md)
 - [Command-line arguments](command-line-arguments.md)
 - [How to read rustdoc output](how-to-read-rustdoc.md)
+- [Rustdoc in-doc settings](in-doc-settings.md)
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/doc/rustdoc/src/SUMMARY.md:6]: I don't completely agree with this classification. For example, rustdoc lints aren't really about "writing rustdoc code", just like clippy isn't about "how to write rust code" but more about "how to improve your code".

Might still be worth it to clean up the rustdoc book structure a bit though. For the time being, I'll move this new chapter as a sub-chapter of "reading rustdoc output".

File: src/doc/rustdoc/src/in-doc-settings.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+browsing to your liking. This chapter will explain what each of them is doing.
+
+As a reminder, the settings menu can be accessed by clicking on the gear icon
+in the upper right.
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/doc/rustdoc/src/in-doc-settings.md]: I can't add this `<i>` unless we add the whole CSS as well and font as well. Not sure it's worth it just for that.

File: src/doc/rustdoc/src/in-doc-settings.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/doc/rustdoc/src/in-doc-settings.md]: I moved the content into the new chapter.

**Comment by GuillaumeGomez** [comment]
Applied suggestions and made this new chapter a new sub-chapter of "how to read rustdoc output".

File: src/doc/rustdoc/src/in-doc-settings.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/doc/rustdoc/src/in-doc-settings.md]: It does but we don't use fontawesome to display the gear. But it's a good idea to include the image here.

[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/doc/rustdoc/src/in-doc-settings.md]: I figured out what was going on: `mdbook` is using the `4.7` version and I was looking at the icon in `5.x` version, which obviously didn't work. Fixed it.

**Review by notriddle — APPROVED** [review]

**Pull request merged**